### DISCUSSION TITLE
Working around default-names removal

### DISCUSSION THREAD

**Discussion by nicolasnoble** [Homebrew/discussions#490] [Tap maintenance and Homebrew development]
So I've been trying to create a formula for building a mips gcc cross compiler for brew, and I'm blocked due to the removal of default-names.

Building gcc *requires* gnu sed being called sed, but then my formula that's trying to use `depends_on "gnu-sed" => "default-names"` isn't working.

Long story short, my build fails constantly, and I have no idea how to solve it given how I need `sed` to be `gsed` at build time given how `default-names` is no longer a thing.

Current work in progress for this effort:

https://github.com/grumpycoders/pcsx-redux/pull/314

Current more global bugs that are likely never going to be solved, describing why it requires sed to be gnu sed, and why I really need "sed" to be "gsed" in my formula:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=62097
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66032
https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=235293

**Comment by carlocab** [comment]
From `brew info gnu-sed`:

```
==> Caveats
GNU "sed" has been installed as "gsed".
If you need to use it as "sed", you can add a "gnubin" directory
to your PATH from your bashrc like:

    PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$PATH"
```

Why don't you add this to your `PATH` at build time?

**Comment by nicolasnoble** [comment]
How do I safely access this path? Is the prefix available somewhere as a variable?

**Reply by carlocab** [reply]
You can try putting this into the `install` block

    ENV.prepend_path "PATH" Formula["gnu-sed"].opt_libexec/"gnubin"

**Reply by nicolasnoble** [reply]
Thanks, this worked. Would've been nice if this was suggested in the deprecation notice of default-names :-)

**Reply by carlocab** [reply]
Seems like a rare enough occurrence to not really need it... also, did setting the environment variable `SED=[/usr/local/bin/]gsed` not work for you?

**Reply by nicolasnoble** [reply]
gcc's build script seemed to ignore it for some reason.
